### PR TITLE
Support NDP proxying on bridge networks

### PR DESCRIPTION
**Pull request by mostolog**
Proxying is enabled via the network label com.docker.network.bridge.ndp_proxy_interface=

Closes https://github.com/docker/libnetwork/issues/1159
Overrides https://github.com/docker/libnetwork/pull/1316

### CODE REVIEW

File: drivers/bridge/setup_ipv6.go [hunk 1]
```diff
@@ -133,17 +132,12 @@ func setupNDPProxying(config *networkConfiguration, i *bridgeInterface) error {
 			logrus.Warnf("Unable to enable NDP default proxying: %v", err)
 		}
 	}
-
-	// Get current NDP all proxying setup
-	ndpProxyDataAll, err := ioutil.ReadFile(ndpProxyConfAll)
```
Comment (aboch) [inline drivers/bridge/setup_ipv6.go]: I see in most of bridge code we first check the flag value, and we set it to 1 only if needed. 
I know we would do only one write if we set it regardless, but I think it is better to be consistent and check first.

**Comment by aboch** [comment]
Thanks @mostolog for carrying this forward.
Thanks for carrying the original commit.
Regarding the extra 3 commits of yours, please squash them into one, and sign it with your id.

**Comment by aboch** [comment]
You now also need a rebase

File: drivers/bridge/setup_ipv6.go (hunk 1 above)
Comment (mostolog) [inline drivers/bridge/setup_ipv6.go]: Do you mean to change https://github.com/mostolog/libnetwork/blob/907282e5195e7bfaf507da0375bcaaf95a618603/drivers/bridge/setup_ipv6.go#L139 ?:

```
    if ndpProxyConfPerm[0] != '1' {
        if err := ioutil.WriteFile(ndpProxyConfPath, []byte{'1', '\n'}, ndpProxyConfPerm); err != nil {
            logrus.Warnf("Unable to enable NDP proxying for interface %s: %v", config.NDPProxyInterface,err)
        }
    }
```

Comment (aboch) [inline drivers/bridge/setup_ipv6.go]: Yeah, to follow same logic we do for the default, for example

https://github.com/mostolog/libnetwork/blob/907282e5195e7bfaf507da0375bcaaf95a618603/drivers/bridge/setup_ipv6.go#L125

First read the value, then set it if not already set.

Comment (mostolog) [inline drivers/bridge/setup_ipv6.go]: It may be solved now

**Comment by aboch** [comment]
Please fetch latest from upstream and rebase your changes. Otherwise we can't merge this.

**Comment by mostolog** [comment]
Already did git _pull --rebase origin master_ and says it's up to date.

**Comment by aboch** [comment]
Do this
```
git fetch <remote>
git rebase <remote>/master
```
(As remote use the one which points to the  github.com/docker/libnetwork repo, check with git remote -v)

**Comment by mostolog** [comment]
Current branch patch-1 is up to date.

File: drivers/bridge/setup_ipv6.go [hunk 2]
```diff
@@ -117,3 +119,31 @@ func setupIPv6Forwarding(config *networkConfiguration, i *bridgeInterface) error
...
+		}
+	}
+
+	// Get interface with proxy_ndp enabled.
+	ndpProxyConfPath := fmt.Sprintf("/proc/sys/net/ipv6/conf/%s/proxy_ndp", config.NDPProxyInterface)
+	if err != nil {
+		return fmt.Errorf("Unable to enable NDP proxying for interface %s: %v", config.NDPProxyInterface,err)
+	}
+	// Enable NDP proxying only if it is not already enabled
+	if ndpProxyConfPerm[0] != '1' {
```
Comment (aboch) [inline drivers/bridge/setup_ipv6.go]: This won't build

**Comment by aboch** [comment]
This has been in non-mergeable state for too long.

I am taking care of the merge conflicts and the modification of `setupNDPProxying()` to only enable the feature on the user specified host interface in #1714

**Pull request closed without merging**